modules/volunteers/sitesupport/volunteersEntry.js

```diff
@@ -0,0 +1,292 @@
+        creditMinutes = creditMinutes % 60;
+      }
+
+      var tag = ' hours';
+      var mtag = ' minutes';
+      if (creditHours == 1) {
+        tag = ' hour';
+      }
+      if (creditMinutes == 1) {
+        mtag = ' minute';
+      }
+      this.totalHours = '';
+      if (creditHours > 0) {
+        this.totalHours = creditHours + tag;
+      }
+      if (creditMinutes > 0) {
+        this.totalHours += ' ' + creditMinutes + mtag ;
+      }
+      this.actualHoursWorked = Number(this.actualHours) + Number(this.actualMinutes) / 60;
+      this.checkHours();
+    },
+    resetForm() {
+      userLookup.clear();
+      this.uid = null;
+      document.getElementById('submit').disabled = true;
+      this.lookupMessage = null;
+      this.departmentsHighlight = null;
+      this.message = null;
+      this.actualHours = 1;
+      this.actualMinutes = 0;
+      this.setEndDate();
+      this.modifier = 1;
+      this.departmentWorked =  this.departments[Number(Object.keys(this.departments)[0])].name;
+      this.timeInvalid = false;
+      this.lookupError = false;
+      this.volunteerName = 'a Volunteer';
+    },
+    handleResult(origin, response) {
+      var e = document.getElementById('userLookup_dropdown');
+      if (!e.classList.contains('UI-hide')) {
+        e.classList.add('UI-hide');
+      }
+      this.timeInvalid = false;
+      this.message = null;
+
+      this.volunteerName = response['First Name'] + ' ' + response['Last Name'];
+      var uid = response.Id;
+      if (response.ConCom && response.ConCom.length > 0) {
+        userLookup.markFailure();
+        document.getElementById('submit').disabled = true;
+        this.lookupMessage = this.volunteerName + ' is on ConCom (' + uid + ')';
+        this.lookupError = true;
+        this.departmentsHighlight = null;
+      } else {
+        userLookup.clearFailure();
+        userLookup.set(uid);
+        document.getElementById('submit').disabled = false;
+        this.lookupMessage = this.volunteerName + ' (' + uid + ')';
+        this.lookupError = false;
+        this.departmentsHighlight = null;
+
+        if ('volunteer' in response) {
+          for (var i = 0, len = response.volunteer.length; i < len; i++) {
+            var dept = response.volunteer[i];
+            if (this.departmentsHighlight == null) {
+              this.departmentsHighlight = new Array(dept['Department ID']);
+            } else if (!this.departmentsHighlight.includes(dept['Department ID'])) {
+              this.departmentsHighlight.push(dept['Department ID']);
+            }
+          }
+        }
+        if ('volunteer' in response) {
+          this.volPast = response.volunteer;
+          this.checkHours();
+        }
+      }
+    },
+    onFail(target, resp, name, code) {
+      userLookup.markFailure();
+      document.getElementById('submit').disabled = true;
+      this.uid = false;
+      this.volunteerName = 'a Volunteer';
+
+      if (code == 404) {
```

Comment:
No handler for 418? :-D

(II had occasion this week to introduce my European padawan to RFC2324)